### ISSUE TITLE
date.getWeekOfYear typed to accept string/number/date but accepts only Date object as parameter

### ISSUE THREAD

**Issue by rathga** [quasarframework/quasar#7759]
**Describe the bug**
``` ts
date.getWeekOfYear('2020-09-10') // fails due to getFullYear does not exist
date.getWeekOfYear(new Date('2020-09-10')) // works fine
```

I guess you just need to wrap the argument inside the getWeekOfYear function before calling .getFullYear in the below code from date.js, or at least change the type definition to only accept Date

``` js
/* ... */
export function getWeekOfYear(date) {
  // Remove time components of date
  var thursday = new Date(date.getFullYear(), date.getMonth(), date.getDate()); // Change date to Thursday same week
/* ... */
'''

**Comment by rstoenescu** [comment]
Hi.

It requires a JS Date object as per design (and docs are accordingly). What's wrong with that? The date utils needs to perform as fast as possible, so we try to avoid useless operations. Just supply a Date object or use the other date methods to create one.

**Comment by rathga** [comment]
That's fine, but the problem is it is TYPED to accept Date | string | number.  If it is designed to use Date object then please change the type definition.  See image.

![image](https://user-images.githubusercontent.com/12701641/92969807-8badde00-f475-11ea-82df-9f4205bcf5b5.png)

**Comment by rstoenescu** [comment]
Fixed the typing. Will be available on Quasar v1.14.
Thanks for the followup!
